packages/assets/path-support.js

```diff
@@ -10,83 +10,4 @@
 
 'use strict';
 
-import type {PackagerAsset} from './registry.js';
-
-const androidScaleSuffix = {
-  '0.75': 'ldpi',
-  '1': 'mdpi',
-  '1.5': 'hdpi',
-  '2': 'xhdpi',
-  '3': 'xxhdpi',
-  '4': 'xxxhdpi',
-};
-
-const ANDROID_BASE_DENSITY = 160;
-
-/**
- * FIXME: using number to represent discrete scale numbers is fragile in essence because of
- * floating point numbers imprecision.
- */
-function getAndroidAssetSuffix(scale: number): string {
-  if (scale.toString() in androidScaleSuffix) {
-    return androidScaleSuffix[scale.toString()];
-  }
-  // NOTE: Android Gradle Plugin does not fully support the nnndpi format.
-  // See https://issuetracker.google.com/issues/72884435
-  if (Number.isFinite(scale) && scale > 0) {
-    return Math.round(scale * ANDROID_BASE_DENSITY) + 'dpi';
-  }
-  throw new Error('no such scale ' + scale.toString());
-}
-
-// See https://developer.android.com/guide/topics/resources/drawable-resource.html
-const drawableFileTypes = new Set([
-  'gif',
-  'jpeg',
-  'jpg',
-  'ktx',
-  'png',
-  'svg',
-  'webp',
-  'xml',
-]);
-
-function getAndroidResourceFolderName(
-  asset: PackagerAsset,
-  scale: number,
-): string | $TEMPORARY$string<'raw'> {
-  if (!drawableFileTypes.has(asset.type)) {
-    return 'raw';
-  }
-  const suffix = getAndroidAssetSuffix(scale);
-  if (!suffix) {
-    throw new Error(
-      "Don't know which android drawable suffix to use for scale: " +
-        scale +
-        '\nAsset: ' +
-        JSON.stringify(asset, null, '\t') +
-        '\nPossible scales are:' +
-        JSON.stringify(androidScaleSuffix, null, '\t'),
-    );
-  }
-  return 'drawable-' + suffix;
-}
-
-function getAndroidResourceIdentifier(asset: PackagerAsset): string {
-  return (getBasePath(asset) + '/' + asset.name)
-    .toLowerCase()
-    .replace(/\//g, '_') // Encode folder structure in file name
-    .replace(/([^a-z0-9_])/g, '') // Remove illegal chars
-    .replace(/^assets_/, ''); // Remove "assets_" prefix
-}
-
-function getBasePath(asset: PackagerAsset): string {
-  const basePath = asset.httpServerLocation;
-  return basePath.startsWith('/') ? basePath.slice(1) : basePath;
-}
-
-module.exports = {
-  getAndroidResourceFolderName,
-  getAndroidResourceIdentifier,
-  getBasePath,
-};
+module.exports = require('./src/path-support');
```

Comment:
To enable running from source, let's add `babel-register` wrapper files for these two entry points. And yeah, actually that means keeping these files around (for Flow's type checker, which sadly doesn't follow `"exports"`).

You can find an example of this pattern in `packages/dev-middleware/`: https://github.com/facebook/react-native/blob/main/packages/dev-middleware/src/index.js.

And the file structure you'll need:

```sh
assets
├── path-support.js.flow # (Flow only) Re-export all types and values
├── registry.js.flow
└── src
    ├── path-support.js # Entry point - babel-register
    ├── path-support.flow.js # Implementation
    ├── registry.js
    └── registry.flow.js
```

Reply:
Thanks, I didn't quite understand what those files were. Makes sense now!

Reply:
So… it looks like `flow-check` is unable to find `.js.flow` files. I had to rename them back to `.js` and fix a bunch of `Dependencies of a @flow strict module must also be @flow strict!` errors. The check seems to pass locally (though I get a bunch of unrelated failures).

Reply:
I'm not sure this pattern works. `babel-register` cannot be found when the package is consumed outside the repository. Do you have any suggestions for how to address this?

Reply:
> `babel-register` cannot be foundwhen the package is consumed outside the repository.

Do you mean you have a setup where another project is consuming a local checkout of the `react-native` repository from source (e.g. via `yarn link`)?

This _should_ work. Happy to debug via chat.